src/librbd/internal.cc

```diff
@@ -1522,6 +1524,190 @@ bool compare_by_name(const child_info_t& c1, const child_info_t& c2)
     return 0;
   }
 
+  int trash_purge(IoCtx& io_ctx, time_t expire_ts, double threshold) {
+  	librbd::NoOpProgressContext pc_ctx;
+  	return trash_purge_with_progress(io_ctx, expire_ts, threshold, pc_ctx);
+  }
+
+  int trash_purge_with_progress(IoCtx& io_ctx, time_t expire_ts,
+          double threshold, ProgressContext& pctx) {
+    auto *cct((CephContext *) io_ctx.cct());
+
+    io_ctx.set_osdmap_full_try();
+    librbd::RBD rbd;
+
+    std::vector<librbd::trash_image_info_t> trash_entries;
+    int r = rbd.trash_list(io_ctx, trash_entries);
```

Comment:
Nit: don't invoke the public librbd API internally -- just use the private version in this file.